sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,547 @@
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import {
+  WebResource,
+  ServiceClient,
+  ServiceClientOptions,
+  HttpOperationResponse,
+  SasServiceClientCredentials,
+  AtomXmlOperationSpec,
+  signingPolicy,
+  userAgentPolicy,
+  logPolicy,
+  proxyPolicy,
+  atomSerializationPolicy,
+  RequestPolicyFactory,
+  ResourceSerializer,
+  URLBuilder,
+  ProxySettings
+} from "@azure/core-http";
+
+import { httpAtomXml } from "./log";
+
+import { QueueResourceSerializer } from "./serializers/queueResourceSerializer";
+import { TopicResourceSerializer } from "./serializers/topicResourceSerializer";
+import { SubscriptionResourceSerializer } from "./serializers/subscriptionResourceSerializer";
+import { RuleResourceSerializer } from "./serializers/ruleResourceSerializer";
+
+/**
+ * All operations return a `Promise<HttpOperationResponse>` object of which the `parsedBody`
+ * is populated with the response in JSON format with following structure:
+ *  {
+ *   response: <actual response>,
+ *   result: <parsed result, if any>,
+ *   error: <error information, if any>
+ *  }
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  endpoint: any;
+
+  queueResourceSerializer: ResourceSerializer;
+  topicResourceSerializer: ResourceSerializer;
+  subscriptionResourceSerializer: ResourceSerializer;
+  ruleResourceSerializer: ResourceSerializer;
+
+  /**
+   * Initializes a new instance of the ServiceBusManagementClient class.
+   * @param credentials Credentials needed for the client to connect to Azure.
+   * @param options The parameter options
+   */
+  constructor(connectionString: string, proxySettings?: ProxySettings) {
+    const connectionStringObj = ServiceBusAtomManagementClient.parseConnectionString(
+      connectionString
+    );
+    const credentials = new SasServiceClientCredentials(
+      connectionStringObj.SharedAccessKeyName,
+      connectionStringObj.SharedAccessKey
+    );
+
+    const requestPolicyFactories: RequestPolicyFactory[] = [];
+    requestPolicyFactories.push(userAgentPolicy());
+    requestPolicyFactories.push(logPolicy(httpAtomXml));
+    requestPolicyFactories.push(atomSerializationPolicy());
+    requestPolicyFactories.push(signingPolicy(credentials));
+
+    if (proxySettings) {
+      requestPolicyFactories.push(proxyPolicy(proxySettings));
+    }
+    const serviceClientOptions: ServiceClientOptions = {
+      requestPolicyFactories: requestPolicyFactories
+    };
+
+    super(credentials, serviceClientOptions);
+    this.queueResourceSerializer = new QueueResourceSerializer();
+    this.topicResourceSerializer = new TopicResourceSerializer();
+    this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
+    this.ruleResourceSerializer = new RuleResourceSerializer();
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*).servicebus.windows.net") || "")[1];
```

Comment:
It looks like you're just grabbing the service bus namespace from the endpoint, not the entire endpoint.

Reply:
The `endpoint` gets used inside `getUrl` to build out url with the full namespace.
`const baseUri = `https://${this.endpoint}.servicebus.windows.net/${path}`;`

Instead, we should parse the entire endpoint like @chradek is suggesting such that in `getUrl()` you only have to
``const baseUri = `https://${this.endpoint}/${path}`;`

This is assuming that `getUrl()` is not expected to always use `servicebus.windows.net`. For example, in the china cloud, you would be using `servicebus.chinacloudapi.cn`. See https://docs.microsoft.com/en-us/azure/china/resources-developer-guide#check-endpoints-in-azure